drivers/sensor/adxl372/adxl372.c

```diff
@@ -0,0 +1,1112 @@
+ * @param status2 - Data stored in the STATUS2 register
+ * @param fifo_entries - Number of valid data samples present in the
+ *			 FIFO buffer (0 to 512)
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int adxl372_get_status(struct device *dev,
+			   u8_t *status1,
+			   u8_t *status2,
+			   u16_t *fifo_entries)
+{
+	u8_t buf[4], length = 1;
+	int ret;
+
+	if (status2) {
+		length++;
+	}
+
+	if (fifo_entries) {
+		length += 2;
+	}
+
+	ret = adxl372_reg_read_multiple(dev, ADXL372_STATUS_1, buf, length);
+
+	*status1 = buf[0];
+
+	if (status2) {
+		*status2 = buf[1];
+	}
+
+	if (fifo_entries) {
+		*fifo_entries = ((buf[2] & 0x3) << 8) | buf[3];
+	}
+
+	return ret;
+}
+
+/**
+ * Software reset.
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int adxl372_reset(struct device *dev)
+{
+	int ret;
+
+	ret = adxl372_set_op_mode(dev, ADXL372_STANDBY);
+	if (ret) {
+		return ret;
+	}
+	/* Writing code 0x52 resets the device */
+	ret = adxl372_reg_write(dev, ADXL372_RESET, ADXL372_RESET_CODE);
+	k_sleep(1000);
+
+	return ret;
+}
+
+/**
+ * Configure the operating parameters for the FIFO.
+ * @param dev - The device structure.
+ * @param mode - FIFO Mode. Specifies FIFO operating mode.
+ *		Accepted values: ADXL372_FIFO_BYPASSED
+ *				 ADXL372_FIFO_STREAMED
+ *				 ADXL372_FIFO_TRIGGERED
+ *				 ADXL372_FIFO_OLD_SAVED
+ * @param format - FIFO Format. Specifies the data is stored in the FIFO buffer.
+ *		Accepted values: ADXL372_XYZ_FIFO
+ *				 ADXL372_X_FIFO
+ *				 ADXL372_Y_FIFO
+ *				 ADXL372_XY_FIFO
+ *				 ADXL372_Z_FIFO
+ *				 ADXL372_XZ_FIFO
+ *				 ADXL372_YZ_FIFO
+ *				 ADXL372_XYZ_PEAK_FIFO
+ * @param fifo_samples - FIFO Samples. Watermark number of FIFO samples that
+ *			triggers a FIFO_FULL condition when reached.
+ *			Values range from 0 to 512.
+
+ * @return 0 in case of success, negative error code otherwise.
+ */
+static int adxl372_configure_fifo(struct device *dev,
+			       enum adxl372_fifo_mode mode,
+			       enum adxl372_fifo_format format,
+			       u16_t fifo_samples)
+{
+	struct adxl372_data *data = dev->driver_data;
+	u8_t fifo_config;
+	int ret;
+
+	if (fifo_samples > 512) {
+		return -EINVAL;
+	}
+
+	/*
+	 * All FIFO modes must be configured while in standby mode.
+	 */
+	ret = adxl372_set_op_mode(dev, ADXL372_STANDBY);
+	if (ret) {
+		return ret;
+	}
+
+	fifo_config = (ADXL372_FIFO_CTL_FORMAT_MODE(format) |
+		       ADXL372_FIFO_CTL_MODE_MODE(mode) |
+		       ADXL372_FIFO_CTL_SAMPLES_MODE(fifo_samples));
+
+	ret = adxl372_reg_write(dev, ADXL372_FIFO_CTL, fifo_config);
+	if (ret) {
+		return ret;
+	}
+	ret = adxl372_reg_write(dev, ADXL372_FIFO_SAMPLES, fifo_samples & 0xFF);
+	if (ret) {
+		return ret;
+	}
+
+	data->fifo_config.fifo_format = format;
+	data->fifo_config.fifo_mode = mode;
+	data->fifo_config.fifo_samples = fifo_samples;
+
+	return 0;
+}
+
+/* Exclude until we have a API to better support this */
+#if 0
```

Comment:
Please don't do this. Remove the FIFO functions from this PR then add them later when we have a way to use them.

Reply:
Removed this dead code and pushed again.

Thanks!